### ISSUE TITLE
Godot 3.4.4 GD Script & C# Mono, AudioEffectRecord Thread warning: A Thread object has been re-started without wait_to_finish() having been called on it

### ISSUE THREAD

**Issue by solarstrings** [godotengine/godot#60712]
### Godot version

3.4.4

### System information

Ubuntu Linux 21.10 64-bit, GeForce RTX 2070 SUPER
Windows 10 64-bit

### Issue description

When running SetRecordingActive(true) the second time on a AudioEffectRecord, it gives the following thread warning:

 A Thread object has been re-started without wait_to_finish() having been called on it. Please do so to ensure correct cleanup of the thread.
    <C++ Source>  core/os/thread.cpp:87 @ start()
    <Stack Trace> :0 @ void Godot.NativeCalls.godot_icall_1_16(IntPtr , IntPtr , Boolean )()

This also happens in the [Mic Recording Demo code](https://github.com/godotengine/godot-demo-projects/tree/3.4-b0d4a7c/audio/mic_record) with GDScript. When you press the Record button the 2nd time, the warning is produced:

![image](https://user-images.githubusercontent.com/2093814/166430726-0c0f78fa-7179-4f54-aeb8-fb0f72295881.png)

### Steps to reproduce

See the reproduction project below for full reproduction demo. 

The minimal needed reproduction is a AudioEffectRecord which has its SetRecordingActive(true) called twise, on a Bus with a record effect on it, in this case, it's pressing the spacebar twise:

```csharp
        if(Input.IsActionJustPressed("spacebar"))
        {     
            _effect.SetRecordingActive(true);
        }
        if(Input.IsActionJustReleased("spacebar"))
        {
            _effect.SetRecordingActive(false);

            // Play the recording
            _rec = _effect.GetRecording();
            _player.Stream = _rec;
            _player.Play();
        }
```

### Minimal reproduction project

[CsharpAudioEffectRecordBug.zip](https://github.com/godotengine/godot/files/8605495/CsharpAudioEffectRecordBug.zip)

**Comment by Calinou** [comment]
@solarstrings Please create a minimal reproduction project in GDScript instead of C# for two reasons:

- Non-Mono builds of Godot are easier to set up and have less moving parts.
- This rules out C# being the cause of the bug.

**Comment by ellenhp** [comment]
When you say `AudioStreamSample` in this issue do you mean `AudioEffectRecord`?

**Comment by solarstrings** [comment]
> When you say `AudioStreamSample` in this issue do you mean `AudioEffectRecord`?

Yes I do. I was late when I uploaded the Issue last night. Updated the text in the Issue

**Comment by solarstrings** [comment]
> @solarstrings Can you reproduce this in GDScript?
> 
> PS: Code blocks should use triple backticks like this (with an optional language name for syntax highlighting):
> 
> ```gdscript code here ```
> 
> I edited your post accordingly, but remember to do this in the future slightly_smiling_face

This is based upon: https://github.com/godotengine/godot-demo-projects/tree/master/audio/mic_record
And the GD Script version of the mic_record runs without causing the thread warning.

**Comment by solarstrings** [comment]
Just tried this with the GDScipt Mic Record Demo, and it gives the same result.
Press and hold the spacebar key to record, and upon release it plays the recording, but produces the 

Copy this code into the Mic Record Demo, and it should produce the Thread warning-
```gdscript
func _process(delta):
	if (Input.is_action_just_pressed("ui_select")):
		effect.set_recording_active(true)
	if (Input.is_action_just_released("ui_select")):
		recording = effect.get_recording()
		effect.set_recording_active(false)
		var data = recording.get_data()
		$AudioStreamPlayer.stream = recording
		$AudioStreamPlayer.play()
	return delta
```

**Comment by solarstrings** [comment]
Also tried the [GDScipt Mic Record Demo,](https://github.com/godotengine/godot-demo-projects/tree/3.4-b0d4a7c/audio/mic_record) in windows, and the bug is here too as well!

**Comment by madmiraal** [comment]
This issue can be closed. It only applied to 3.x, and it was fixed with #62135.
